eipw-lint/src/lib.rs

```diff
@@ -419,12 +419,24 @@ pub fn default_lints_enum() -> impl Iterator<Item = (&'static str, DefaultLint<&
                 message: "proposals must be referenced with the form `EIP-N` (not `EIPN` or `EIP N`)",
             }),
         ),
+        (
+            "markdown-link-eip",
+            MarkdownLinkEip {
+                pattern: markdown::LinkEip(r"eip-([^.]*)\.md(#.+)?$")
+            }
+        ),
         (
             "markdown-link-first",
             MarkdownLinkFirst {
                 pattern: markdown::LinkFirst(r"(?i)(?:eip|erc)-[0-9]+"),
             }
         ),
+        (
+            "markdown-link-other",
+            MarkdownLinkOther {
+                pattern: markdown::LinkOther(r"^(EIP|ERC)-(\d+)\s*\S*$")
```

Comment:
Should this be case insensitive (`(?i)`)?

Reply:
I am not sure about it. EIP-1 says the text should be like `EIP-N`. Does it mean upper case or not? I can add case insensitive if it is so. Then [eip-25]() will work for links. Is it ok or it must be [EIP-25]() (upper case only)?

Reply:
I think we have the other lint to check for incorrectly cased references. So even if `[eip-1](./eip-1.md)` passes `markdown-link-other`, it'll fail the other lint.

Or at least I think it will :thinking: